src/lib/components/Menu/Menu.js

```diff
@@ -122,6 +136,105 @@ class Menu extends Component {
     if (this.props.showMap && this.props.showMap !== prevProps.showMap && this.props.menuScroll) {
       this.menuWrapper.current.scrollTop = this.props.menuScroll.scrollTop;
     }
+
+    const { sharedLayers } = this.state;
+    if (sharedLayers.filter(l => !l.isCatOpen).length) {
+      /** If there are any shared layers whose category we haven't open,
+       * open them
+       */
+      this.openCategoryForSharedLayers(sharedLayers);
+    }
+  }
+
+  /**
+   * Open category for which each of the shared layers falls under
+   */
+  openCategoryForSharedLayers(layersToOpen) {
+    const { categories } = this.props;
+    // const { sharedLayers } = this.state;
+
+    if (layersToOpen && categories) {
+      layersToOpen
+        .filter(l => !l.isCatOpen)
+        .forEach(sharedLayer => {
+          let i = 0;
+          /**
+           * A layer belongs to only one category. So if we found its
+           * category, use this flag to break from the loop
+           */
```

Comment:
> a layer belongs to only one category

Is this necessarily true? Maybe that's how it's been but I don't think this is a hard requirement 🤔